core/src/rpc.ts

```diff
@@ -28,7 +29,10 @@ export const setupRpcUrlAndGetNetworks = async function (
     isDev = false
 ): Promise<{ networks: NetworkConfig[]; defaultNetwork: string; defaultChainId: string }> {
     if (!rpcUrl) {
-        throw new Error(`No RPC_URL env variable was provided`);
+        throw new Error(`Invalid RPC URL`);
+    }
+    if (getNetworks()) {
+        resetNetworks();
     }
```

Comment:
What does this try to accomplish? Why do you need to reset networks in the first place? Why is it related to the execution of the `getNetworks`?

Reply:
Ok, but why do you need to check `if (getNetworks())` first before resetting it?
If you _are required to check it_, why it's not part of the `resetNetworks` function itself?